app/components/clicker_game.rb

```diff
@@ -0,0 +1,51 @@
+class ClickerGame < ViewComponent::Base
+  include Motion::Component
+
+  attr_reader :key, :player, :score, :game
+
+  map_motion :click
+
+  PLAYERS = %i[
+    dragon fly pigeon gull eagle robin flea mosquito airplane
+    rocket flamingo raptor falcon owl squirrel dino zebra
+    unicorn puma giraffe pikachu chair glasses sorcerer wizard
+    knight squire cauldron sparrow finch penguin otter]
+
+  def initialize(key:)
+    @key = key
+    @player = PLAYERS.sample.to_s
+    @game = { player => 0 }
+
+    stream_from game_channel, :player_click
+
+    track_score
+  end
+
+  def click(event)
+    amt = event.target.data[:amt] || 1
```

Comment:
You are correct. If you read the data attribute off of the `Event`, it is just going to include whatever is in the DOM.

The only thing the user can't modify is `data-motion-state` (and even then it won't have an effect if the component is already mounted). If they modify `data-motion-key`, the might loose some nested state when an outer component renders, but nothing will fail to validate.

If you want to prevent the user from cheating, you need to verify that that amount is unlocked in the motion.